File: 3. Baekjoon/1929. 소수 구하기.py
import sys

input = sys.stdin.readline


if __name__ == "__main__":
    # 숫자의 범위가 크므로 에라토스테네스의 체를 이용해야 한다는 직감
    M, N = map(int, input().split())

    # 에라토스테네스 체 선언, 0과 1은 소수가 아니므로 기본값 False
    arr = [False, False] + [True] * (N - 1)
    cnt = 2

    while cnt <= N:
        # 하나씩 지워나감
        # 2부터 시작해서, 해당 수는 소수로 두고 그 수의 배수들을 모두 False처리
        if arr[cnt]:
            for j in range(cnt + cnt, N + 1, cnt):
                if arr[j]:
                    arr[j] = False
        cnt += 1

    for i in range(M, N + 1):
        if arr[i]:
            print(i)

File: 3. Baekjoon/1929. 소수 구하기.cpp
#include <iostream>
#include <vector>

using namespace std;

typedef vector<bool> BoolArray;

int main() {
    int M, N;
    cin >> M >> N;

    // 에라토스테네스의 체 배열 초기화, 0과 1은 소수가 아니므로 기본값 false
    BoolArray arr(N + 1, true);
    arr[0] = arr[1] = false;

    for (int cnt = 2; cnt <= N; cnt++) {
        if (arr[cnt]) {
            for (int j = cnt + cnt; j <= N; j += cnt) {
                arr[j] = false;
            }
        }
    }

    // M부터 N까지의 소수를 출력
    for (int i = M; i <= N; i++) {
        if (arr[i]) {
            cout << i << "\n";
        }
    }

    return 0;
}
